azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/AzureAccount.java

```diff
@@ -5,38 +5,124 @@
-import com.microsoft.azure.toolkit.lib.auth.model.AccountEntity;
+import com.microsoft.azure.toolkit.lib.auth.exception.LoginFailureException;
+import com.microsoft.azure.toolkit.lib.auth.model.AuthConfiguration;
+import com.microsoft.azure.toolkit.lib.auth.model.AuthType;
+import com.microsoft.azure.toolkit.lib.auth.util.AzureEnvironmentUtils;
+import org.apache.commons.lang3.ObjectUtils;
 
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class AzureAccount implements AzureService {
-
     private Account account;
 
-    public List<Account> getAvailableAccounts() {
-        return null;
-    }
+    public static void main(String[] args) throws LoginFailureException {
+        AzureAccount az = Azure.az(AzureAccount.class);
+        for (Account account : az.accounts()) {
+            System.out.println("Got: " + account.getMethod());
+            System.out.println("Got: " + account.isAvailable());
+            System.out.println("Got: " + account.isAuthenticated());
+            az.login(account);
+        }
 
-    /**
-     * @return account if authenticated
-     * @throws AzureToolkitAuthenticationException if failed to authenticate
-     */
-    public Account login(AccountEntity entity) throws AzureToolkitAuthenticationException {
-        final Account account = new Account(); // TODO: login and create Account instance;
-        this.account = account;
-        return this.account;
     }
 
     /**
-     * @return account if authenticated
-     * @throws AzureToolkitAuthenticationException if not authenticated
+     * @return the current account
+     * @throws AzureToolkitAuthenticationException if not initialized
      */
     public Account account() throws AzureToolkitAuthenticationException {
         return Optional.ofNullable(this.account)
-            .filter(Account::isAuthenticated)
-            .orElseThrow(() -> new AzureToolkitAuthenticationException("Account is not authentication."));
+                .orElseThrow(() -> new AzureToolkitAuthenticationException("Account is not initialized."));
+    }
+
+    public List<Account> accounts() {
+        return buildAccountMap(AzureEnvironment.AZURE).values().stream().collect(Collectors.toList());
+    }
+
+    public AzureAccount login(Account targetAccount) throws LoginFailureException {
+        account = targetAccount;
+        account.authenticate();
+        return this;
+    }
+
+    public AzureAccount login(AuthType type) {
+        AuthConfiguration auth = new AuthConfiguration();
+        auth.setType(type);
+        loginWithAuthConfiguration(auth);
+        return this;
+    }
+
+    public Account servicePrincipalAccount(AuthConfiguration config) {
+        return null;
+    }
+
+    public void login(AuthConfiguration auth) throws LoginFailureException {
```

Comment:
`login(@Nonnull AuthConfiguration auth)`
the caller should guarantee `auth` is not null.